app/medInria/medRegistrationWorkspace.cpp

```diff
@@ -130,123 +111,75 @@ bool medRegistrationWorkspace::isUsable()
     return (tbFactory->toolBoxesFromCategory("Registration").size()!=0);
 }
 
-void medRegistrationWorkspace::updateFromMovingContainer()
+void medRegistrationWorkspace::updateFromFixedContainer()
 {
-    if(!d->selectorToolBox)
-        return;
-
-    if(!d->movingContainer->view())
-    {
-        medAbstractLayeredView* fuseView  = dynamic_cast<medAbstractLayeredView*>(d->fuseContainer->view());
-        if(fuseView)
-        {
-            if(fuseView->layer(d->selectorToolBox->movingData()) == 0)
-            {
-                d->fuseContainer->removeView();
-                d->fuseContainer->addData(d->selectorToolBox->fixedData());
-            }
-            else
-                fuseView->removeLayer(1);
-
-        }
-
-        d->selectorToolBox->setMovingData(NULL);
-        return;
-    }
-
-    medAbstractLayeredView* movingView  = dynamic_cast<medAbstractLayeredView*>(d->movingContainer->view());
-    if(!movingView)
-    {
-        qWarning() << "Non layered view are not suported yet in registration workspace.";
-        return;
-    }
-
-    medAbstractData *movingData = movingView->layerData(movingView->currentLayer());
-
-    medAbstractLayeredView* fuseView  = dynamic_cast<medAbstractLayeredView*>(d->fuseContainer->view());
-    if(fuseView)
-        fuseView->removeData(movingData);
-
-    d->fuseContainer->addData(movingData);
-    fuseView  = dynamic_cast<medAbstractLayeredView*>(d->fuseContainer->view());
-
-    if(movingData)
-    {
-        d->viewGroup->addImpactedView(movingView);
-        d->viewGroup->addImpactedView(fuseView);
+    updateFromContainers(0);
+}
 
-        d->movingLayerGroup->addImpactedlayer(movingView, movingData);
-        d->movingLayerGroup->addImpactedlayer(fuseView, movingData);
-    }
-    if (!d->selectorToolBox->setMovingData(movingData))
-    {
-        // delete the view because something failed at some point
-        d->viewGroup->removeImpactedView(movingView);
-        d->movingLayerGroup->removeImpactedlayer(movingView, movingData);
-        d->movingLayerGroup->removeImpactedlayer(fuseView, movingData);
-        movingView->deleteLater();
-    }
+void medRegistrationWorkspace::updateFromMovingContainer()
+{
+    updateFromContainers(1);
 }
 
-void medRegistrationWorkspace::updateFromFixedContainer()
+void medRegistrationWorkspace::updateFromContainers(int containerIndex)
```

Comment:
Technically this function is updating from one container, not from several as it's name suggests.